framework/src/logger/logger.ts

```diff
@@ -107,6 +107,66 @@ class ConsoleLog {
 	}
 }
 
+class FileLog {
+	private readonly _stream: fs.WriteStream;
+
+	public constructor(filePath: string) {
+		this._stream = fs.createWriteStream(filePath, { flags: 'a', encoding: 'utf8' });
+	}
+
+	public write(rec: LogObject): void {
+		const {
+			time,
+			level,
+			name,
+			msg,
+			module: moduleName,
+			err,
+			hostname,
+			pid,
+			src,
+			v,
+			...others
+		} = rec;
+		const log: Record<string, unknown> = {
+			time,
+			level,
+			name,
+			msg,
+			hostname,
+			pid,
+			module: moduleName ?? 'unknown',
+		};
+		const otherKeys = Object.entries(others);
+		if (otherKeys.length > 0) {
+			const meta = otherKeys.reduce<Record<string, unknown>>((prev, [key, value]) => {
```

Comment:
This does not cover all possible cases. 

1. Nested objects with `buffer` or `bigint` values. 
2. Arrays on first level or deep level

I am not sure if such nested objects are not used within SDK. But on a higher perspective we are exposing the logger from the `Application` instance as `app.logger` so SDK user can log anything with. So our logger implementation should be very generic to handle all such cases.

Reply:
I think there is a trade off of how nested that logs needs to go vs performance.
For the console log also it only checks the first level. This is focusing on the fixing the bug that it logs the Buffer JSON on our usage.

If we want to support more nested level, we should create new issue as a feature issue

Reply:
https://github.com/LiskHQ/lisk-sdk/issues/5778